src/main/java/com/oltpbenchmark/benchmarks/templated/procedures/GenericQuery.java

```diff
@@ -36,21 +36,37 @@ public abstract class GenericQuery extends Procedure {
 
     /** Execution method with parameters. */
     public void run(Connection conn, List<Object> params) throws SQLException {
-        try (PreparedStatement stmt = getStatement(conn, params); ResultSet rs = stmt.executeQuery()) {
-            while (rs.next()) {
-                //do nothing
+
+        try (PreparedStatement stmt = getStatement(conn, params)) {
+            try (ResultSet rs = stmt.executeQuery()) {
```

Comment:
Can you restructure this to something like this instead:
https://stackoverflow.com/a/70024146
Basically, use [`execute()`](https://docs.oracle.com/javase/8/docs/api/java/sql/PreparedStatement.html#execute--) to explicitly check if it should have had a result set and then handle each case appropriately rather than implicitly relying on `executeQuery()` to throw an exception when it encounters an insert/update/delete statement.